src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -1514,4 +1514,12 @@ internal static void AddPrimaryErrorMessage(CodeElement currentElement, string n
         }
         CrawlTree(currentElement, x => AddPrimaryErrorMessage(x, name, type, asProperty));
     }
+    protected static void DeduplicateErrorMappings(CodeElement codeElement)
```

Comment:
Out of curiosity would it also be valid to perform the deduplication in the builder in this scenario? (assuming all downstream languages support the XXX code) https://github.com/microsoft/kiota/blob/e5ab5e2bc4dcfd0532f0a4bc10b1d3ee9037ed43/src/Kiota.Builder/KiotaBuilder.cs#L1288

In my head, the only reason is there possible "information loss" if one of the languages does not support the `XXX` and thus needs the two values.

Reply:
Yes. The reason why I'm doing it in the refiners is because we might want to do it per language. But we could move it to the builder in the future, hence why most of the logic is actually in code method.

Reply:
Thanks for clearing that up.